posthog/api/csp.py

```diff
@@ -0,0 +1,146 @@
+    report_to_data["script-sample"] = escape(report_to_data.get("sample"))
+    properties = {
+        "report_type": data.get("type"),
+        "$current_url": report_to_data.get("documentURL") or report_to_data.get("document-uri") or data.get("url"),
+        "document_url": report_to_data.get("documentURL") or report_to_data.get("document-uri"),
+        "referrer": report_to_data.get("referrer"),
+        "violated_directive": report_to_data.get("effectiveDirective")
+        or report_to_data.get("violated-directive"),  # Inferring from effectiveDirective
+        "effective_directive": report_to_data.get("effectiveDirective"),
+        "original_policy": report_to_data.get("originalPolicy"),
+        "disposition": report_to_data.get("disposition"),
+        "blocked_url": report_to_data.get("blockedURL") or report_to_data.get("blocked-uri"),
+        "line_number": report_to_data.get("lineNumber"),
+        "column_number": report_to_data.get("columnNumber"),
+        "source_file": report_to_data.get("sourceFile"),
+        "status_code": report_to_data.get("statusCode"),
+        "script_sample": report_to_data.get("sample"),
+        "user_agent": user_agent,
+        # Keep the raw report for debugging
+        "raw_report": data,
+    }
+    return properties
+
+
+def parse_properties(data: dict) -> dict:
+    if "csp-report" in data:
+        return parse_report_uri(data)
+    elif "type" in data and data["type"] == "csp-violation":
+        return parse_report_to(data)
+    else:
+        raise ValueError("Invalid CSP report")
+
+
+def process_csp_report(request):
+    """
+    Process a Content Security Policy (CSP) report from a browser.
+
+    Takes the incoming CSP report JSON, formats it as a PostHog event,
+    and returns it for ingestion through the regular event pipeline.
+
+    Returns:
+        tuple: (csp_report, error_response)
+            - csp_report: The formatted CSP report as a PostHog event, or None if processing failed
+            - error_response: An error response to return to the client if processing failed, or None if successful
+    """
+    try:
+        # If by any chance we got this far and this is not looking like a CSP report, keep the ingestion pipeline working as it was
+        # we don't want to return an error here to avoid breaking the ingestion pipeline
+        if request.content_type != "application/csp-report" and request.content_type != "application/reports+json":
+            return None, None
+
+        csp_data = json.loads(request.body)
+
+        distinct_id = request.GET.get("distinct_id") or str(uuid7())
+        session_id = request.GET.get("session_id") or str(uuid7())
+        version = request.GET.get("v") or "unknown"
+
+        properties = parse_properties(csp_data)
+
+        return {
+            "event": "$csp_violation",
+            "distinct_id": distinct_id,
+            "properties": {"$session_id": session_id, "csp_version": version, **properties},
```

Comment:
```suggestion
            "properties": {"$session_id": session_id, "csp_version": version, $process_person_profile: False, **properties},
```

Reply:
should we allow person processing if you send a distinct id?

Reply:
ok, that seems the desired behaviour to me... and safer than setting true 👍 (which we can always do later if nececssary)